test/integration/apiserver/admissionplugins/shootvalidator/shootvalidator_test.go

```diff
@@ -0,0 +1,321 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shootvalidator_test
+
+import (
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+var _ = Describe("ShootValidator tests", func() {
+	var (
+		seed              *gardencorev1beta1.Seed
+		shoot             *gardencorev1beta1.Shoot
+		testSecret        *corev1.Secret
+		testSecretBinding *gardencorev1beta1.SecretBinding
+
+		clusterRole *rbacv1.ClusterRole
+		roleBinding *rbacv1.RoleBinding
+
+		user           *envtest.AuthenticatedUser
+		userTestClient client.Client
+		userName       string
+
+		err error
+	)
+
+	BeforeEach(func() {
+		By("creating SecretBinding")
```

Comment:
If I see it correctly, you don't need to create Secret, SecretBinding, and Seed for every test individually.
Instead, you could create them in `BeforeSuite` as well :)